src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediateStatement.java

```diff
@@ -47,23 +51,39 @@
   private static final char SINGLE_QUOTE = '\'';
 
   public IntermediateStatement(String sql, Connection connection) throws SQLException {
-    this();
     this.sql = sql;
     this.statements = parseStatements(sql);
     this.command = parseCommand(sql);
     this.connection = connection;
     this.statement = connection.createStatement();
+    // Note: This determines the result type based on the first statement in the SQL statement. That
+    // means that it assumes that if this is a batch of statements, all the statements in the batch
+    // will have the same type of result (that is; they are all DML statements, all DDL statements,
+    // all queries, etc.). That is a safe assumption for now, as PgAdapter currently only supports
+    // all-DML and all-DDL batches.
+    this.resultType = determineResultType(sql);
   }
 
-  protected IntermediateStatement() {
-    this.executed = false;
-    this.exception = null;
-    this.resultType = null;
-    this.hasMoreData = false;
-    this.statementResult = null;
-    this.updateCount = null;
```

Comment:
Removed as it is void. A constructor call will always automatically initialize properties to their default values (e.g. false for booleans, null for reference types, ...)